server/src/main/java/io/whitefox/persistence/memory/InMemoryStorageManager.java

```diff
@@ -43,6 +45,31 @@ public InMemoryStorageManager(List<PShare> shares) {
         shares.stream().collect(Collectors.toMap(PShare::name, Function.identity())));
   }
 
+  @Override
+  public CompletionStage<Optional<Integer>> getTableVersion(
```

Comment:
Yeah that is because we return A Table and not a Table from the StorageManager, we can change that and let the service be in charge of doing the conversion. Actually it makes a lot of sense, you can make that refactor if you feel like it.